lib/galaxy/datatypes/data.py

```diff
@@ -404,9 +406,8 @@ def display_info(self, dataset):
             if info.find( '\n' ) >= 0:
                 info = info.replace( '\n', '<br/>' )
 
-            # Convert to unicode to display non-ascii characters.
-            if not isinstance(info, unicode):
-                info = unicode( info, 'utf-8')
+            if not isinstance(info, text_type):
+                info = unicodify( info, 'utf-8' )
```

Comment:
If `info` is already in non-utf-8 unicode, `unicodify` will not modify it, you can try with `unicodify(text_type(b'ciao', 'iso-8859-1'))` in both Python 2 and 3.